### ISSUE TITLE
npx hardhat run support

### ISSUE THREAD

**Issue by lomonoshka** [zemse/hardhat-tracer#28]
Hey, thanks for the grate plugin!
Would it be possible to add support of --trace to hardhat run command?

**Comment by zemse** [comment]
I'll explore if this can be supported. But for now this workaround can be used:

```ts
import hre from 'hardhat'; 
import { wrapHardhatProvider } from "hardhat-tracer/dist/src/wrapper";

wrapHardhatProvider(hre);
hre.tracer.enabled = true;
hre.tracer.verbosity = 3; // prints all txs, use 1 if you need only failed printings

async function main() {
  ...
}
```

**Comment by AlexBlack772** [comment]
I need this feature too, It would be very usefull.

Maybe this is a limitation of hre/ethers but provided workaround show traces only for successfull calls. If call ended up with a revert, no traces are shown. but with npx hardhat trace --hash <hash of failed tx> all traces are shown

**Comment by zemse** [comment]
> Maybe this is a limitation of hre/ethers but provided workaround show traces only for successfull calls

Yeah this was the case in V1 because it relied on debug_traceTransaction. But in v2 it hijacks the VM to get the execution info and now you do not need to do this. `--trace` would print all successful and failed while `--traceError` would only print the failed transactions or calls for you. To use v2 you can just install latest version `npm i hardhat-tracer@latest`.
